src/command_modules/azure-cli-network/azure/cli/command_modules/network/_help.py

```diff
@@ -667,11 +672,26 @@
 helps['network asg'] = """
     type: group
     short-summary: Manage application security groups.
+    long-summary: You can configure network security as a natural extension of an application’s structure, ASG allows
+     you to group virtual machines and define network security policies based on those groups. You can specify an
+      application security group as the source and destination in a NSG security rule. For more information
+       visit https://docs.microsoft.com/en-us/azure/virtual-network/create-network-security-group-preview
 """
 
 helps['network asg create'] = """
     type: command
     short-summary: Create an application security group.
+    long-summary: You can configure network security as a natural extension of an application’s structure, ASG
+     allows you to group virtual machines and define network security policies based on those groups. You can
+      specify an application security group as the source and destination in a NSG security rule. For more
+       information visit https://docs.microsoft.com/en-us/azure/virtual-network/create-network-security-group-preview
+    parameters:
+        - name: --name -n
```

Comment:
Is this parameter required, and is there a way to tag a parameter as required within the docs itself?

P.S. We still have the ongoing "some options are in docs, some are not" issue.

Reply:
The CLI determines when a parameter is required. The help YAML doesn't determine that.

As for the second point, mostly it is "FEW options are in docs, MOST are not."  However, the help YAML is the final source, so it will override anything that comes from a different source.  Feel free to reach out to me if you want to discuss that point further.